File: core/test/org.ebayopensource.vjet.test.core.jstojava/testFiles/access/returnStatement/AccessReturn1Tester.js
vjo.ctype('access.returnStatement.AccessReturn1Tester')
.protos({

//> public int getIsle(int val1, String val2)
getIsle : function (val1, val2)
{
if (val1 == 0)
{
	return 0;
}
else if (val1 == 1)
{
return 0;
}
else if (val1 == 2)
{
return 0;
}
else if(val2 == "String"){return 0;}
else if(val2 === "string"){return 0;}
else if(val1 > 0){return 0;}
else if(val1 < 30){return 0;}
else if(val1 >= 0){return 0;}
else if(val1 <= 30){return 0;}
else if(val1 == 0){return 0;}
else if(val1 != 0){return 0;}
else return 0;
}
})
.endType();

File: core/test/org.ebayopensource.vjet.test.core.jstojava/src/org/ebayopensource/dsf/jst/validation/vjo/access/returncheck/AccessReturn1Tester.java
/*******************************************************************************
 * Copyright (c) 2005-2011 eBay Inc.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 *******************************************************************************/
/* 
 * $Id: AccessReturnCorrect1Tester.java, May 26, 2009, 7:58:31 PM, liama. Exp$
 *
 * Copyright (c) 2006-2009 Ebay Technologies. All Rights Reserved.
 * This software program and documentation are copyrighted by Ebay 
 * Technologies.
 */
package org.ebayopensource.dsf.jst.validation.vjo.access.returncheck;




import java.util.List;

import org.ebayopensource.dsf.jsgen.shared.validation.vjo.VjoSemanticProblem;
import org.ebayopensource.dsf.jst.validation.vjo.VjoValidationBaseTester;
import org.junit.Before;
import org.junit.Test;




/**
 * Access return correct 1 tester
 *
 * @author <a href="mailto:liama@ebay.com">liama</a>
 * @since JDK 1.5
 */
//@ModuleInfo(value="DsfPrebuild",subModuleId="JsToJava")
//@Category( { P1, FAST, UNIT })
public class AccessReturn1Tester extends VjoValidationBaseTester {

    @Before
    public void setUp() {
        expectProblems.clear();
    }

    
    @Test
    //@Category( { P1, FAST, UNIT })
    //@Description("Test return statement")
    public void testAccesVisible1() {
        List<VjoSemanticProblem> problems = getVjoSemanticProblem(
                "access.returnStatement.","AccessReturn1Tester.js", this
                        .getClass());
        assertProblemEquals(expectProblems, problems);
    }
    
    @Test
    //@Category( { P1, FAST, UNIT })
    //@Description("Test return statement")
    public void testAccesVisible2() {
        expectProblems.clear();
        List<VjoSemanticProblem> problems = getVjoSemanticProblem(
                "access.returnStatement.","AccessReturn3.js", this
                        .getClass());
        assertProblemEquals(expectProblems, problems);
    }
}
